lib/rubocop/cop/rails/file_path.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # @example
+      #  # bad
+      #  Rails.root.join('app/models/goober')
+      #  File.join(Rails.root, 'app/models/goober')
+      #  "#{Rails.root}/app/models/goober"
+      #
+      #  # good
+      #  Rails.root.join('app', 'models', 'goober')
+      class FilePath < Cop
+        MSG = 'Please use `Rails.root.join(\'path\', \'to\')` instead.'.freeze
+
+        def_node_search :file_join_nodes, <<-PATTERN
+          (send (const nil :File) :join ...)
+        PATTERN
+
+        def_node_search :file_join_nodes?, <<-PATTERN
+          (send (const nil :File) :join ...)
+        PATTERN
+
+        def_node_search :rails_root_nodes?, <<-PATTERN
+          (send (const nil :Rails) :root)
+        PATTERN
+
+        def_node_search :rails_root_join_nodes, <<-PATTERN
+          (send (send (const nil :Rails) :root) :join ...)
+        PATTERN
+
+        def on_dstr(node)
+          return unless rails_root_nodes?(node)
+          offense(node)
+        end
+
+        def on_send(node)
+          file_join_with_rails_root(node)
+          rails_root_join_with_slash_separated_path(node)
+        end
+
+        private
+
+        def file_join_with_rails_root(node)
```

Comment:
I'd name this `check_for_file_join_with_rails_root` or something else. Or maybe you could extract a separate predicate method from this one to simplify the code?